[batch infer] Update batch inference template to use RayLLMBatch (#343)
anyscale/templates: templates/batch-llm/README.md

Some models will require you to input your [Hugging Face user access token](https://huggingface.co/docs/hub/en/security-tokens). This will be used to authenticate/download the model and **is required for official LLaMA, Mistral, and Gemma models**. You can use one of the other models which don't require a token if you don't have access to this model (for example, `mlabonne/NeuralHermes-2.5-Mistral-7B`).

Run the following cell to start the authentication flow. A VS Code overlay will appear and prompt you to enter your Hugging Face token if your selected model requires authentication. If you are using a model that does not require a token, you can skip this step. For this example, we will be using the `meta-llama/Meta-Llama-3.1-8B-Instruct` model, which requires a token.

```python
# Prompts the user for Hugging Face token if required by the model.
from util.utils import prompt_for_hugging_face_token
HF_TOKEN = prompt_for_hugging_face_token("meta-llama/Meta-Llama-3.1-8B-Instruct")
```

In this example, we will be using the `meta-llama/Meta-Llama-3.1-8B-Instruct` model, 

> We will also need to define a model yaml associated with the model we want to use to configure the compute resources, engine arguments and other inference engine specific parameters. For more details on the the model configs, see the [API doc](https://docs.anyscale.com/llms/serving/guides/bring_any_model/) on bringing your own models.

Reviewer: ```suggestion
We will also need to define a yaml configuration file associated with the model we want to use to configure the compute resources, engine arguments and other inference engine specific parameters. For more details on the the model configs, see the [API doc](https://docs.anyscale.com/llms/serving/guides/bring_any_model/) on bringing your own models.
```

```python

from rayllm_batch import init_engine_from_config
# Read the model configs from the path.
model_config_path = "configs/llama-3.1-8b-a10g.yaml"

# One could potentially override the engine configs by passing in a dictionary here.
override = {"runtime_env": {"env_vars": {"HF_TOKEN": HF_TOKEN}}} # Override Ray's runtime env to include the Hugging Face token. Ray is being used under the hood to orchestrate the inference pipeline.
engine_config = init_engine_from_config(config=model_config_path, override=override)

```

## Step 3: Run the batch inference through RayLLMBatch

With the workload and model configs defined, we can now run the batch inference through RayLLMBatch.

```python

from rayllm_batch import RayLLMBatch

batch = RayLLMBatch(
    engine_config=engine_config,
    workload=workload,

>     # Specify the batch size for inference. Set the batch size to as large as possible without running out of memory.
>     # If you encounter out-of-memory errors, decreasing batch_size may help. 

Reviewer: ```suggestion
    # Specify the batch size for inference. Set the batch size to as large as possible without running out of memory to maximize the throughput.
    # Meanwhile, a reasonable batch size can offer better fault tolerance.
```

    batch_size=None,
    # Set the number of replicas to use for the inference. Each replica will run one instance of inference pipeline.
    num_replicas=1,
)

> # This will runs until completion.
> ds = batch.run()

Reviewer: ```suggestion
# This will runs until completion. If you specify output_path=..., then
# the results will be written to local disk (local://) or AWS S3 (s3://).
# In this example, we only keep results in memory for demo purpose.
ds = batch.run()
```

> # Read the results
> gen_texts = [r["generated_text"] for r in ds.take_all()]

Reviewer: ```suggestion
# Peek the first 3 results
gen_texts = [r["generated_text"] for r in ds.take_all(3)]
```
